### PR TITLE
Issue 2608 tags with require

### PR DESCRIPTION
**Pull request by crtrott**
This fixes issue #2608. Essentially the last argument checkup in the argument list for policies will always be interpreted as "WorkTag" but if there gets an extra void inserted everything falls apart. This change just removes voids from the list.

### CODE REVIEW

File: core/src/impl/Kokkos_AnalyzePolicy.hpp [hunk 1]
```diff
@@ -120,6 +120,17 @@ struct SetWorkTag {
                                 typename PolicyBase::work_item_property>;
 };
 
+// Partial specialization to drop void arguments
```
Comment (dalg24) [inline core/src/impl/Kokkos_AnalyzePolicy.hpp]: Why did you chose that over `!std::is_void<T> ? SetWorkTag<Base, T> : Base` in `AnalyzePolicy`?

File: core/unit_test/standalone/Makefile [hunk 2]
```diff
@@ -9,7 +9,6 @@ ifndef KOKKOS_PATH
 endif
 
 SRC = $(wildcard $(MAKEFILE_PATH)*.cpp)
-SRC += $(MAKEFILE_PATH)/../TestStackTrace.cpp
```
[the remark this answers is not in the document]
Comment (crtrott) [reply core/unit_test/standalone/Makefile]: I did!

**Comment by brian-kelley** [comment]
@crtrott @srajama1 Could this be merged into the 3.0 release candidate? Exawind and EMPIRE want CUDA graph support in the kokkos-kernels smoothers.

Also, the testing failures in travis don't seem to be related to these changes.

**Comment by crtrott** [comment]
Hi, we are not gonna merge this in for 3.0. But we can apply this thing as a hotfix to trillions later.

**Comment by srajama1** [comment]
@crtrott Is this already in develop ? Note that we are starting to test the smoothers needed by Exawind (March milestone). This changes are needed for one of the smoothers to be competitive.

**Comment by brian-kelley** [comment]
@srajama1 It is, and I have the gauss-seidel running correctly with cuda graphs. I have only run that with Kokkos develop (not the 3.0 candidate) but I expect this is the only thing that needs to be hotfixed to Trilinos.

**Comment by brian-kelley** [comment]
@srajama1 So Trilinos can be taken care of - the issue I see now is that I can't merge that stuff into kokkos-kernels develop because it's still tested against kokkos master, right? Then the kokkos-kernels snapshot will be ahead of develop which is something to avoid.

**Pull request merged**